src/charm.py

```diff
@@ -84,14 +83,18 @@ def _configure(self, event):
             return
 
         try:
-            self._set_alerts()
-            self._set_mimir_config()
-            self._set_pebble_layer()
+            restart = any(
+                [
+                    self._set_alerts(),
+                    self._set_mimir_config(),
+                    self._set_pebble_layer(),
+                ]
+            )
         except BlockedStatusError as e:
             self.unit.status = BlockedStatus(e.message)
             return
 
-        if self._restart:
+        if restart:
             self._container.restart(self._name)
```

Comment:
`restart` can raise, so perhaps, for robustness, it could be advantageous to:
```python
try:
   if any(...):
      restart()
except ...:
    ...
else:
    logger.info(...)
```